cli/pkg/kctrl/cmd/app/init/interfaces.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package init
+
+import (
+	v1alpha12 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Build interface {
+	Save() error
+	GetAppSpec() *v1alpha12.AppSpec
+	SetAppSpec(*v1alpha12.AppSpec)
+	GetObjectMeta() *metav1.ObjectMeta
+	SetObjectMeta(*metav1.ObjectMeta)
+	SetExport(export *[]Export)
+	GetExport() *[]Export
```

Comment:
We typically don't use getters and setters in Go. We can have `AppSpec` instead of `GetAppSpec` and so on.
Also, we are using the `.` notation to get the required variables from Build in a couple of places which kind of defeats the purpose of having a method for just _get_.